### PR TITLE
Build models dynamically from data instead of manually

### PR DESCRIPTION
**Pull request by gburgett**
Given a store full of Contentful entries downloaded from the Sync API, we now can dynamically generate both a set of Model objects which automatically resolve links, and a GraphQL schema which can be used to query the linked data.

Example:
```ruby
main_menu = WCC::Contentful::Model.find('FNlqULSV0sOy4IoGmyWOW')

# assert
expect(main_menu.hamburger).to be_instance_of(WCC::Contentful::Menu)
expect(main_menu.hamburger.first_group[0]).to be_instance_of(WCC::Contentful::MenuItem)
expect(main_menu.hamburger.first_group[0].link).to be_instance_of(WCC::Contentful::Page)
expect(main_menu.hamburger.first_group[0].link.title).to eq('About')
```

```ruby
query_string = '{
    menu: ContentfulMenu(id: "FNlqULSV0sOy4IoGmyWOW") {
      hamburger {
        firstGroup {
          link {
            title
          }
        }
      }
    }
}'
result = schema.execute(query_string)

# assert
expect(result.to_h['errors']).to be_nil
expect(result.dig('data', 'menu', 'hamburger', 'firstGroup', 1, 'link', 'title'))
      .to eq('Mission')
```

### CODE REVIEW

File: lib/wcc/contentful/model.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+# frozen_string_literal: true
+
+module WCC::Contentful
+  class Model
+    class << self
+      include Helpers
```
Comment (jpowell) [inline lib/wcc/contentful/model.rb]: equivalent to `extend`

File: wcc-contentful.gemspec [hunk 2]
```diff
@@ -1,30 +1,43 @@
...
-  spec.require_paths = ["lib"]
-
-  spec.add_development_dependency "bundler", "~> 1.16"
-  spec.add_development_dependency "rake", "~> 10.0"
-  spec.add_development_dependency "rspec", "~> 3.0"
-  spec.add_development_dependency "webmock", "~> 3.0"
-  spec.add_development_dependency "vcr", "~> 4.0"
-  spec.add_dependency "contentful_model", "~> 0.2.0"
+  spec.require_paths = ['lib']
+
```
[the remark this answers is not in the document]
Comment (jpowell) [reply wcc-contentful.gemspec]: theporch-live zsh 2018-02-28 16-48-50
Case and point - trying to use on Porch which specifies `2.2.6` as its version

[the remark this answers is not in the document]
Comment (gburgett) [reply wcc-contentful.gemspec]: I'll get to that in the next PR - gonna build on top of this for https://zube.io/watermarkchurch/development/c/2059

File: lib/wcc/contentful/content_type_indexer.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        :Int
+      when 'Number'
+        :Float
+      when 'Date'
+        :DateTime
+      when 'Boolean'
+        :Boolean
+      when 'Object'
+        :Json
+      when 'Location'
```
Comment (jpowell) [inline lib/wcc/contentful/content_type_indexer.rb]: still `Location`?

Comment (jpowell) [reply lib/wcc/contentful/content_type_indexer.rb]: I thought you hadn't updated your intermediate fixture yet...

File: lib/wcc/contentful.rb [hunk 4]
```diff
@@ -32,12 +35,17 @@ def self.configure_contentful_model
         end
       end
     end
+
+    def self.init!
+      raise ArgumentError, 'Please first call WCC:Contentful.Configure!' if configuration.nil?
```
Comment (jpowell) [inline lib/wcc/contentful.rb]: no implementation yet?

File: lib/wcc/contentful/content_type_indexer.rb (hunk 3 above)
Comment (gburgett) [reply lib/wcc/contentful/content_type_indexer.rb]: 'Location' is Contentful's name for this data type.  Here is where we translate that to 'coordinates'.

File: lib/wcc/contentful/configuration.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  def sync_store
+    @sync_store = SYNC_STORES[@sync_store].call(self) if @sync_store.is_a? Symbol
+    @sync_store ||= Store::MemoryStore.new
+  end
+
+  def client
+    @client ||=
+      if defined?(ContentfulModel)
+        configuration.configure_contentful_model
+        @client = ContentfulModel::Base.client
```
Comment (jpowell) [inline lib/wcc/contentful/configuration.rb]: no need to set here

**Pull request merged**